### PR TITLE
FetchTxs and unblindTxs functions in wallet.ts

### LINKED ISSUE #56: [SDK] add fetchAndUnblindTxs methods
At the moment there is no way to fetch all previous transactions ﻿for and address and group them for asset. 
We should provide a method like fetchAndUnblindTxs(address: string, blindPrivKey: string, url: string)

### PR DESCRIPTION
**Pull request by louisinger**
**This PR adds a new exported function in `wallet.ts` that fetch the txs from the explorer endpoint and unblind it**

The `fetchAndUnblindTxs` do:
1. Retrieve ALL the transactions associated to an address given as parameter from an Esplora API.
2. Get the hex of the txs from the txids.
3. Construct `Transaction` object.
4. Unblind the transaction outputs with a given blinding private key.

**QUESTION**: In my opinion we could let the frontend handles the unblinding step because it depends of the use case. For now, we provide only one private blinding key because we find txs associated with one address. However, in reality, a transaction is composed by several wallet addresses (what about the change output for instance?). 

_What I propose:_ Split the `fetchAndUnblindTxs` in two functions --> a `fetchTxs(address: string, explorerUrl: string): Transaction[]` and a `unblindOutputsInTransaction(tx: Transaction, blindingPrivateKeys: string[]): Transaction`. Could be more "flexible".

@tiero please review

it closes #56

### CODE REVIEW

**Comment by tiero** [comment]
I agreee with the proposal of splitting the function in two methods. Go ahead

File: README.md [hunk 1]
```diff
@@ -1,36 +1,37 @@
 # 🛠 tdex-sdk
+
```
Comment (tiero) [inline README.md:2]: I guess you have some sort of markdown linter. Please deactivate

**Review by tiero — CHANGES_REQUESTED** [review]
Not sure that returning `liquidjs` data structure as result of the methods is something useful.

Like for utxos, we have an UtxoInterface wich basically reflect what you would expect from calling esplora like API.

I would recreate the explorer's interface and return that, so clients/frontends will interact with is with common type without installing/learning how to use liquidjs

File: README.md (hunk 1 above)
Comment (louisinger) [reply README.md:2]: done

**Comment by louisinger** [comment]
> Not sure that returning `liquidjs` data structure as result of the methods is something useful.
> 
> Like for utxos, we have an UtxoInterface wich basically reflect what you would expect from calling esplora like API.
> 
> I would recreate the explorer's interface and return that, so clients/frontends will interact with is with common type without installing/learning how to use liquidjs

I've used liquidjs data structure `Transaction` because the `Tx` returned by Esplora endpoint does not return proofs (only the commitments) of outputs. However we needs `rangeProof` and `surjectionProof` to unblind the Output. 
Also, I thought that by returning a Transaction, it would be easier to manipulate/modify the transaction using liquidjs-lib. The user doesn't necessarily need to understand liquidjs-lib: the `Transaction` type is an interface like any other (different but on the same level as the one proposed by Esplora). 
PS: In fact, `UtxoInterface` use a liquidjs-lib data structure: `TxOutput`

**Comment by tiero** [comment]
> I've used liquidjs data structure `Transaction` because the `Tx` returned by Esplora endpoint does not return proofs (only the commitments) of outputs. However we needs `rangeProof` and `surjectionProof` to unblind the Output.

It's ok to create custom interface if we need.

In case we want use Transaction instead, we shoule to export it in the index and index.web, like we do with `networks`

**Review by tiero — APPROVED** [review]

**Pull request merged**